framework/src/base/NonlinearSystemBase.C

```diff
@@ -333,6 +333,8 @@ NonlinearSystemBase::addBoundaryCondition(const std::string & bc_name, const std
   const std::set<BoundaryID> & boundary_ids = bc->boundaryIDs();
   _vars[tid].addBoundaryVar(boundary_ids, &bc->variable());
 
+  checkEigenBoundaryConditionConsistency(bc, parameters);
+
```

Comment:
I do not think this is correct.

1. if I had a normal FEProblem and I added a homogeneous Dirichlet BC, I would get an error that it not possible, but it is perfectly fine. Can you verify that on your branch? I do not think we have a test with homogeneous BCs.
2. doing this check every time we add BC is probably ok, but I though we would do it in `checkProblemIntegrity` that is in `FEProblemBase`. That method should become virtual and it should call something like `_nl.checkIntegrity()` (a new method on our systems) where you would go over the objects from the warehouses and did your check.There is already a integrity checking action in MOOSE, so we should use that when ever possible...